tests/unit-tests/scene/test_surface_stack.cpp

```diff
@@ -1458,4 +1438,21 @@ TEST_F(SurfaceStack, multiple_surfaces_can_be_swapped)
             SceneElementForStream(stub_buffer_stream2)));
 
     Mock::VerifyAndClearExpectations(&observer);
-}
\ No newline at end of file
+}
+
+TEST_F(SurfaceStack, observer_is_notified_on_lock)
+{
+    auto observer = std::make_shared<MockSessionLockObserver>();
+    EXPECT_CALL(*observer, on_lock()).Times(1);
+    stack.register_interest(observer);
+    stack.lock();
+}
+
+TEST_F(SurfaceStack, observer_is_notified_on_unlock)
+{
+    auto observer = std::make_shared<MockSessionLockObserver>();
+    EXPECT_CALL(*observer, on_unlock()).Times(1);
+    stack.lock();
+    stack.register_interest(observer);
+    stack.unlock();
+}
```

Comment:
I'm sure we've discussed more scenarios to test: multiple `lock()s` for instance.